### PR TITLE
Cleanup Sendable conformances and @preconcurrency imports to build as strict concurrency clean

### PR DESCRIPTION
**Pull request by phausler**
This corrects a number of stray requirements for `Sendable` on iterators and adjusts a few cases of no-longer-needed `@preconcurrency` annotations. This now brings us to a fully clean `-strict-concurrency=complete` build.

### CODE REVIEW

File: Sources/AsyncAlgorithms/AsyncThrowingChannel.swift [hunk 1]
```diff
@@ -9,7 +9,7 @@
 //
 //===----------------------------------------------------------------------===//
 
-import OrderedCollections
+@preconcurrency import OrderedCollections
```
Comment (phausler) [inline Sources/AsyncAlgorithms/AsyncThrowingChannel.swift:12]: I wonder if we should mark these as `@_implementationOnly`

Comment (FranzBusch) [reply Sources/AsyncAlgorithms/AsyncThrowingChannel.swift:12]: IMO, I consider it a best practice nowadays to mark every import as `@_implementationOnly` it really should be the default.

Comment (phausler) [reply Sources/AsyncAlgorithms/AsyncThrowingChannel.swift:12]: I will follow-up with an additional audit of those

**Review by FranzBusch — APPROVED** [review]
LGTM!

**Pull request merged**